src/transformers/trainer.py

```diff
@@ -1908,7 +1908,10 @@ def _save_tpu(self, output_dir: Optional[str] = None):
         else:
             self.model.save_pretrained(output_dir, save_config=self.args.should_save, save_function=xm.save)
         if self.tokenizer is not None and self.args.should_save:
-            self.tokenizer.save_pretrained(output_dir)
+            if self.tokenizer.is_fast:
+                self.tokenizer.save_pretrained(output_dir, legacy_format=False)
+            else:
+                self.tokenizer.save_pretrained(output_dir, legacy_format=True)
```

Comment:
This fix proposal changes the behavior of the `trainer`. In this proposal, only the files needed to rebuild the currently used tokenizer are saved.

In other words, if you use the fast version of a tokenizer the files of the slow version - if they could be saved without error - are not saved anymore.